src/Microsoft.VisualStudio.Composition/Configuration/Formatters/MetadataObjectFormatter.cs

```diff
@@ -151,45 +153,80 @@ public void Serialize(ref MessagePackWriter messagePackWriter, object? value, Me
         }
     }
 
-    public object? Deserialize(ref MessagePackReader reader, MessagePackSerializerOptions options)
+    public ImmutableDictionary<string, object?> DeserializeMetadataObjects(ref MessagePackReader reader, MessagePackSerializerOptions options, int count)
```

Comment:
💅🏼 Avoid using `public` on members that are not intended to be public API. 
In this case, you're in an `internal` type so ultimately this API is internal, making `public` here misleading. During a code review, public API impact merits more careful review than internal changes, so seeing this sets me up to immediately ask "Why is this *public*?" only to later realize "oh, it's not _really_ public, because it's in an `internal` type", so it's a time waste.

The only reason you should ever have `public` members on an `internal` type is when the C# language requires it, which is true for operator overloads and interface implementing members.

Reply:
I don't see this method used in this class at all. It serves only the class from which you removed the code that serves as this method body. How is this valuable? It increases the code churn in the code review, so there should be some value to it or IMO it should be reverted.